servicetalk-http-api/src/main/java/io/servicetalk/http/api/StrategyInfluencerAwareConversions.java

```diff
@@ -30,76 +33,98 @@ static StreamingHttpServiceFilterFactory toConditionalServiceFilterFactory(
         requireNonNull(predicate);
         requireNonNull(original);
 
-        if (original instanceof HttpExecutionStrategyInfluencer) {
-            HttpExecutionStrategyInfluencer influencer = (HttpExecutionStrategyInfluencer) original;
-            return new StrategyInfluencingStreamingServiceFilterFactory() {
-                @Override
-                public StreamingHttpServiceFilter create(final StreamingHttpService service) {
-                    return new ConditionalHttpServiceFilter(predicate, original.create(service), service);
-                }
-
-                @Override
-                public HttpExecutionStrategy influenceStrategy(final HttpExecutionStrategy strategy) {
-                    return influencer.influenceStrategy(strategy);
-                }
-            };
-        }
-        return service -> new ConditionalHttpServiceFilter(predicate, original.create(service), service);
+        HttpExecutionStrategy requiredOffloads =
+                StrategyInfluencerAwareConversions.requiredOffloads(original,
+                        HttpExecutionStrategyInfluencer.noInfluence());
+
+        return requiredOffloads.hasOffloads() ?
+                new StrategyInfluencingStreamingServiceFilterFactory() {
+
+                    @Override
+                    public HttpExecutionStrategy requiredOffloads() {
+                        return requiredOffloads;
+                    }
+
+                    @Override
+                    public StreamingHttpServiceFilter create(final StreamingHttpService service) {
+                        return new ConditionalHttpServiceFilter(predicate, original.create(service), service);
+                    }
+                } :
+                service -> new ConditionalHttpServiceFilter(predicate, original.create(service), service);
```

Comment:
if the returned value isn't a strategy influencer, couldn't it be treated as default strategy (offload all) later on?